src/twisted/newsfragments/10273.bugfix

```diff
@@ -0,0 +1,2 @@
+Deprecated ``twisted.python.threading.ThreadPool.currentThread()`` in favor of ``threading.current_thread()``.
```

Comment:
Does just recommending use of the underlying stdlib function seem better?

Reply:
Yes, that's strictly better.